clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/processor/JustificationProcessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 Lumberjack authors (see AUTHORS file)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.abcxyz.lumberjack.auditlogclient.processor;
+
+import com.abcxyz.jvs.JvsClient;
+import com.abcxyz.lumberjack.v1alpha1.AuditLogRequest;
+import com.google.api.client.util.StringUtils;
+import com.google.cloud.audit.AuditLog;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import org.apache.commons.codec.binary.Base64;
+import com.google.inject.Inject;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor(onConstructor = @__({@Inject}))
+public class JustificationProcessor {
+  private static final String JUSTIFICATION_LOG_METADATA_KEY = "justification";
+  private final JvsClient jvs;
+
+  /**
+   * Validates the given {@code jvsToken} and populates the {@link AuditLogRequest} with the
+   * justification info.
+   *
+   * @param jvsToken A JWT produced by JVS.
+   * @param auditLogRequest Audit log request to be processed.
+   * @return The {@link AuditLogRequest} with justification filled.
+   * @throws LogProcessingException when it fails to populate the justification info.
+   */
+  public AuditLogRequest process(String jvsToken, AuditLogRequest auditLogRequest)
+      throws LogProcessingException {
+    if (!auditLogRequest.hasPayload()) {
+      throw new IllegalArgumentException("audit log request doesn't have payload");
+    }
+
+    AuditLogRequest.Builder auditLogRequestBuilder = auditLogRequest.toBuilder();
+    AuditLog.Builder auditLogBuilder = auditLogRequest.getPayload().toBuilder();
+    this.setLogJustification(jvsToken, auditLogBuilder);
+
+    return auditLogRequestBuilder.setPayload(auditLogBuilder.build()).build();
+  }
+
+  /**
+   * Validates the given {@code jvsToken} and sets the justification info in the given
+   * {@code auditLogBuilder}.
+   *
+   * @param jvsToken A JWT produced by JVS.
+   * @param auditLogBuilder Audit log builder.
+   * @throws LogProcessingException when it fails to set the justification info.
+   */
+  public void setLogJustification(String jvsToken, AuditLog.Builder auditLogBuilder)
```

Comment:
It is atypical within java to have a method that modifies the inputs. Usually any results should be returned. Therefore, the more idiomatic way would be to have this return the metadata, and then allow the calling function to set it in the AuditLog.Builder

Reply:
That's _exactly_ what I'm trying to avoid. I don't want the caller to worry about where to set the justification.
Maybe I should return the same `auditLogBuilder`?

Reply:
I believe that this should take a string and return a struct. One property of java is that it does not create copies of arguments passed in, so modifying an arugment modifies the underlying object and all future references to that object, even outside the method. Therefore, it has become standard practice to either explictly make or treat inputs as final if possible, and return modified values.

Reply:
Discussed offline with @myurtoglu  who has Java readability, he seemed to be on the same page.

Reply:
Sure. But that's against my original purpose, which is to make a lib to properly set the justification in the _right_ place in the log. How should achieve that?

Reply:
So the process method is fine. I think the process method should take the output of this method (which will return a struct) and set it in the audit log.

Reply:
So, here https://github.com/abcxyz/lumberjack/pull/273/files#diff-ec13edea722c4098caa7f8b168cb91a5701fdeeb83f84a43cd3d1895b8d589adR56 take the struct and set it in the auditlog builder.

Reply:
I thought about that. But then in https://github.com/abcxyz/lumberjack/pull/273/files#diff-1386f01992f8e23e4621a9698030e5661cbc9368288e50a4018158d86d27820e, it only gives an `auditLogBuilder` to set justification. How should I handle that?

Reply:
As discussed offline, I changed the signature to return a copy of the builder. Thanks for the help!